build/Dockerfile

```diff
@@ -1,4 +1,14 @@
-# Build the manager binary
+# Copyright (c) 2021 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+#   Red Hat, Inc. - initial API and implementation
+#
+
 FROM quay.io/libpod/golang:1.13 as builder
```

Comment:
I think we can drop rhel Dockerfile and just proceed with `registry.access.redhat.com/ubi8-minimal:8.3-298` base here.
golang image is very limited and I did not manage to make simple `if` stuff working.

Reply:
+1 for dropping the libpod variant. 

But right now I'm fighting bigger problems... 

https://issues.redhat.com/browse/CRW-1671?focusedCommentId=16035277&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16035277

```
2021-04-08 20:06:10,537 - atomic_reactor.plugins.imagebuilder - INFO - go: github.com/devfile/api/v2@v2.0.0-20210212161736-283b0c54946e: Get https://proxy.golang.org/github.com/devfile/api/v2/@v/v2.0.0-20210212161736-283b0c54946e.mod: dial tcp 172.217.15.81:443: connect: connection refused
```

Maybe the go.mod isn't fully up to date?

Reply:
We had issues at one point when building without goproxy because openshift/api deleted all their tags and so dependencies couldn't be resolved. The UBI images IIRC use `GOPROXY=direct`, which was a problem (hence the `RUN go env GOPROXY` line below). See https://github.com/devfile/devworkspace-operator/issues/166

Reply:
welll.... if I run `go mod tidy` in the 0.2.x branch I get a LOT of changes:

```
$➔ gd
diff --git a/codeready-workspaces-devworkspace-controller/go.mod b/codeready-workspaces-devworkspace-controller/go.mod
index ef738a0..49c8990 100644
--- a/codeready-workspaces-devworkspace-controller/go.mod
+++ b/codeready-workspaces-devworkspace-controller/go.mod
@@ -3,28 +3,19 @@ module github.com/devfile/devworkspace-operator
 go 1.13
 
 require (
-       github.com/devfile/api/v2 v2.0.0-20210212161736-283b0c54946e
-       github.com/go-logr/logr v0.1.0
-       github.com/google/go-cmp v0.5.0
-       github.com/google/uuid v1.1.1
-       github.com/kr/pretty v0.2.0 // indirect
-       github.com/onsi/ginkgo v1.12.1
-       github.com/onsi/gomega v1.10.1
-       github.com/openshift/api v0.0.0-20200205133042-34f0ec8dab87
-       github.com/redhat-cop/operator-utils v0.1.0
-       github.com/stretchr/testify v1.6.1
-       go.uber.org/zap v1.16.0 // indirect
-       golang.org/x/crypto v0.0.0-20200622213623-75b288015ac9
-       golang.org/x/lint v0.0.0-20191125180803-fdd1cda4f05f // indirect
-       golang.org/x/net v0.0.0-20200625001655-4c5254603344 // indirect
-       golang.org/x/sys v0.0.0-20200625212154-ddb9806d33ae // indirect
-       golang.org/x/tools v0.0.0-20200403190813-44a64ad78b9b // indirect
-       google.golang.org/protobuf v1.25.0 // indirect
-       gopkg.in/yaml.v3 v3.0.0-20200605160147-a5ece683394c // indirect
-       k8s.io/api v0.18.8
-       k8s.io/apiextensions-apiserver v0.18.8 // indirect
-       k8s.io/apimachinery v0.18.8
-       k8s.io/client-go v0.18.8
-       sigs.k8s.io/controller-runtime v0.6.3
+       github.com/devfile/api/v2 v2.0.0
+       github.com/go-logr/logr v0.4.0
+       github.com/google/go-cmp v0.5.5
+       github.com/google/uuid v1.2.0
+       github.com/onsi/ginkgo v1.16.1
+       github.com/onsi/gomega v1.11.0
+       github.com/openshift/api v0.0.0-20210409143810-a99ffa1cac67
+       github.com/redhat-cop/operator-utils v1.1.2
+       github.com/stretchr/testify v1.7.0
+       golang.org/x/crypto v0.0.0-20210322153248-0c34fe9e7dc2
+       k8s.io/api v0.21.0
+       k8s.io/apimachinery v0.21.0
+       k8s.io/client-go v11.0.0+incompatible
+       sigs.k8s.io/controller-runtime v0.8.3
        sigs.k8s.io/yaml v1.2.0
 )
```

What I'm trying to figure out is what I need to run to bootstrap the build such that everything I need is already in a vendor/ folder that I can commit along with the rest of the sources. 

Right now I'm doing this, which works locally but fails to resolve all the deps in Brew:

```
COPY go.mod go.mod
COPY go.sum go.sum
#RUN go mod download # commented out, can't do this in Brew

# compile workspace controller binaries, then webhook binaries
RUN export ARCH="$(uname -m)" && if [[ ${ARCH} == "x86_64" ]]; then export ARCH="amd64"; elif [[ ${ARCH} == "aarch64" ]]; then export ARCH="arm64"; fi && \
  CGO_ENABLED=0 GOOS=linux GOARCH=${ARCH} GO111MODULE=on go build -v -x \
  -a -o _output/bin/devworkspace-controller \
  -gcflags all=-trimpath=/ \
  -asmflags all=-trimpath=/ \
  main.go
RUN export ARCH="$(uname -m)" && if [[ ${ARCH} == "x86_64" ]]; then export ARCH="amd64"; elif [[ ${ARCH} == "aarch64" ]]; then export ARCH="arm64"; fi && \
  CGO_ENABLED=0 GOOS=linux GOARCH=${ARCH} GO111MODULE=on go build -v -x \
  -o _output/bin/webhook-server \
  -gcflags all=-trimpath=/ \
  -asmflags all=-trimpath=/ \
  webhook/main.go
```

Reply:
note that go mod tidy also complains about a bunch of missing stuff:
```
$➔ go mod tidy || echo "failed!"
go: finding github.com/googleapis/gnostic/OpenAPIv2 latest
go: finding sigs.k8s.io/controller-runtime/pkg latest
go: finding sigs.k8s.io/controller-runtime/pkg/runtime/log latest
go: finding sigs.k8s.io/controller-runtime/pkg/runtime latest
go: finding k8s.io/api/batch latest
go: finding k8s.io/api/auditregistration/v1alpha1 latest
go: finding k8s.io/api/settings latest
go: finding k8s.io/api/settings/v1alpha1 latest
go: finding k8s.io/api/auditregistration latest
go: finding k8s.io/api/batch/v2alpha1 latest
go: finding k8s.io/client-go/metadata latest
github.com/devfile/devworkspace-operator/pkg/webhook/kubernetes imports
	sigs.k8s.io/controller-runtime/pkg/runtime/log: module sigs.k8s.io/controller-runtime@latest found (v0.8.3), but does not contain package sigs.k8s.io/controller-runtime/pkg/runtime/log
github.com/devfile/devworkspace-operator imports
	k8s.io/client-go/kubernetes/scheme imports
	k8s.io/api/auditregistration/v1alpha1: module k8s.io/api@latest found (v0.21.0), but does not contain package k8s.io/api/auditregistration/v1alpha1
github.com/devfile/devworkspace-operator imports
	k8s.io/client-go/kubernetes/scheme imports
	k8s.io/api/batch/v2alpha1: module k8s.io/api@latest found (v0.21.0), but does not contain package k8s.io/api/batch/v2alpha1
github.com/devfile/devworkspace-operator imports
	k8s.io/client-go/kubernetes/scheme imports
	k8s.io/api/settings/v1alpha1: module k8s.io/api@latest found (v0.21.0), but does not contain package k8s.io/api/settings/v1alpha1
github.com/devfile/devworkspace-operator/controllers/controller/devworkspacerouting imports
	sigs.k8s.io/controller-runtime/pkg/client imports
	k8s.io/client-go/metadata: module k8s.io/client-go@latest found (v11.0.0+incompatible), but does not contain package k8s.io/client-go/metadata
github.com/devfile/devworkspace-operator/pkg/infrastructure imports
	k8s.io/client-go/discovery imports
	github.com/googleapis/gnostic/OpenAPIv2: module github.com/googleapis/gnostic@latest found (v0.5.4), but does not contain package github.com/googleapis/gnostic/OpenAPIv2
```

Reply:
ok, I've fixed it. Just needed to use -mod=vendor (and actually commit the 42M vendor folder into pkgs.devel).